pytest-pixl/src/pytest_pixl/dicom.py

```diff
@@ -14,308 +14,114 @@
-    ds.LowRRValue = "0"
-    ds.HighRRValue = "0"
-    ds.IntervalsAcquired = "0"
-    ds.IntervalsRejected = "0"
-    ds.HeartRate = "0"
-    ds.ReconstructionDiameter = "256.0"
-    ds.ReceiveCoilName = "MULTI COIL"
-    ds.AcquisitionMatrix = [0, 256, 256, 0]
-    ds.InPlanePhaseEncodingDirection = "ROW"
-    ds.FlipAngle = "8.0"
-    ds.SAR = "0.01492713019251"
-    ds.dBdt = "51.8182716369628"
-    ds.B1rms = 0.6147387027740479
-    ds.PatientPosition = "HFS"
-    ds.AcquisitionDuration = 115.6185302734375
-    ds.DiffusionBValue = 0.0
-    ds.DiffusionGradientOrientation = [0.0, 0.0, 0.0]
-    ds.StudyInstanceUID = "1.3.46.670589.11.38023.5.0.14068.2023012517090166000"
-    ds.SeriesInstanceUID = "1.3.46.670589.11.38023.5.0.7404.2023012517551898153"
-    ds.StudyID = "727549740"
-    ds.SeriesNumber = "901"
-    ds.AcquisitionNumber = "9"
-    ds.InstanceNumber = "1"
     ds.ImagePositionPatient = list(image_position_patient)
-    ds.ImageOrientationPatient = [
-        -0.0065220510587,
-        0.99990475177764,
-        -0.0121616190299,
-        0.05751159414649,
-        -0.0117666730657,
-        -0.9982755184173,
-    ]
-    ds.FrameOfReferenceUID = "1.3.46.670589.11.38023.5.0.12268.2023012517152734001"
-    ds.Laterality = ""
-    ds.TemporalPositionIdentifier = "1"
-    ds.NumberOfTemporalPositions = "1"
-    ds.PositionReferenceIndicator = ""
     ds.SliceLocation = slice_location
-    ds.SamplesPerPixel = 1
-    ds.PhotometricInterpretation = "MONOCHROME2"
-    ds.Rows = 256
-    ds.Columns = 256
-    ds.PixelSpacing = [1, 1]
-    ds.BitsAllocated = 16
-    ds.BitsStored = 12
-    ds.HighBit = 11
-    ds.PixelRepresentation = 0
     ds.WindowCenter = window_centre
     ds.WindowWidth = window_width
-    ds.RescaleIntercept = "0.0"
-    ds.RescaleSlope = "5.47863247863247"
-    ds.RescaleType = "normalized"
-    ds.RequestingPhysician = ""
-    ds.RequestingService = ""
-    ds.RequestedProcedureDescription = ""
-    ds.RequestedContrastAgent = ""
-    ds.StudyComments = ""
-    ds.SpecialNeeds = ""
-    ds.PatientState = ""
-    ds.ScheduledPerformingPhysicianName = ""
-    ds.PerformedStationAETitle = "STATIONAE"
-    ds.PerformedStationName = ""
-    ds.PerformedLocation = ""
-    ds.PerformedProcedureStepStartDate = "20230101"
-    ds.PerformedProcedureStepStartTime = "170902"
-    ds.PerformedProcedureStepEndDate = "20230101"
-    ds.PerformedProcedureStepEndTime = "170902"
-    ds.PerformedProcedureStepStatus = ""
-    ds.PerformedProcedureStepID = "727549740"
-    ds.PerformedProcedureStepDescription = "Step Description"
-    ds.PerformedProcedureTypeDescription = ""
+    ds.PixelData = pixel_data.tobytes()
 
-    # Performed Protocol Code Sequence
-    performed_protocol_code_sequence = Sequence()
-    ds.PerformedProtocolCodeSequence = performed_protocol_code_sequence
+    # Handle any additional tags
+    for key, value in kwargs.items():
+        # Check if tag is DICOOM compliant
+        if dictionary_has_tag(key):
```

Comment:
I don't think this will work in practice.  For example, [this document](https://github.com/rordenlab/dcm2niix/tree/master/Philips#diffusion-direction) tells us that `(0x2005, 0x1412)` is important for diffusion images from a Philips scanner but that tag isn't present in pydicom's dictionary of known tags.  I'd expect the same to be true for other private space tags.

Reply:
I see, originally I didn't put this check in, but that allows setting any arbitrary tag name, as in `ds.blablbabla = "what"`. But I guess since this is just to generate test data, that's not such a big problem?

Reply:
Or, how about having a separate function `add_private_tags()` that uses [`pydicom.Dataset.private_block()`](https://pydicom.github.io/pydicom/stable/old/private_data_elements.html#setting-private-data-elements-with-pydicom) to add private tags?